### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by masterDen** [mrniko/netty-socketio#30]
Could you write documentation (guide,  etc) for implementation SocketIOServer listeners?

**Comment by niffrig** [comment]
Additionally interface documentation for things like JsonSupport.  The expectations for namespaces and arrays are unclear.  What are the expectations of lifecycle and how are they expected to scale?  It appears that a JsonSupport instance is effectively meant to be request scoped but that happens as a detail of implementation in the default JacksonJsonSupport.

**Comment by AlexCzar** [comment]
Also I couldn't find any info on how to make SpringAnnotationScanner work. Declaring it like this, does nothing:
```groovy

@Bean(destroyMethod = "stop")
def socketServer() {
    new SocketIOServer(new com.corundumstudio.socketio.Configuration())
}

@Bean
def socketIoConfigurer() {
    new SpringAnnotationScanner(socketServer())
}
```

**Comment by mrniko** [comment]
@AlexCzar i have almost the same config as you wrote and it works. Is socketServer available as bean?

**Comment by AlexCzar** [comment]
Yes it is. For now I add all the handler in the `static void main()` like this:
```groovy

def app = SpringApplication.run(Bootstrap, args)
def ioServer = app.getBean(SocketIOServer)
ioServer.addListeners(MyHandler)
ioServer.start()
```
